### PR TITLE
Stricter assert for Hilbert integration test

### PR DESCRIPTION
**Pull request by vtomole**
Remove simple test and refine the `assert of https://github.com/Infleqtion/client-superstaq/pull/777

### CODE REVIEW

File: cirq-superstaq/cirq_superstaq/daily_integration_test.py [hunk 1]
```diff
@@ -286,13 +287,24 @@ def test_submit_to_provider_simulators(target: str, service: css.Service) -> Non
...
+        css.ParallelRGate(np.pi / 2, 0.0, 24).on(*qubits),
+        cirq.rz(np.pi).on(qubits[2]),
+        css.ParallelRGate(-np.pi / 2, 0.0, 24).on(*qubits),
+        cirq.measure(*qubits),
+    )
 
-    job = service.create_job(circuit=circuit, repetitions=100, target="cq_hilbert_qpu")
-    assert sum(job.counts().values()) == 100
+    job = service.create_job(
+        circuit=circuit, repetitions=400, verbatim=True, route=False, target=target
```
Comment (richrines1) [inline cirq-superstaq/cirq_superstaq/daily_integration_test.py:303]: do we need this many shots? it seems like 100 should still be enough for this to pass consistently (though also not sure if there's any reason to use fewer)

**Review by richrines1 — APPROVED** [review]
one question but lgtm!

File: cirq-superstaq/cirq_superstaq/daily_integration_test.py (hunk 1 above)
Comment (vtomole) [reply cirq-superstaq/cirq_superstaq/daily_integration_test.py]: Changed to `100`

**Comment by vtomole** [comment]
[Passes after 10 minutes](https://github.com/Infleqtion/client-superstaq/actions/runs/6242256478) when not reserved.

**Pull request merged**